src/components/navbar/navBar.tsx

```diff
@@ -17,15 +17,21 @@ import Logo from "../../assets/react.svg";
 import { useState } from "react";
 import pages from "../../datas/pages.tsx";
 import { useNavigate } from "react-router-dom";
+import { request } from "../../utils";
 
 export default function NavBar() {
   const [open, setOpen] = useState(false);
 
   const navigate = useNavigate();
   const handleDrawerOpen = () => {
-    setOpen(!open);
+    setOpen((prevState) => !prevState);
   };
 
+  const handleLogout = async () =>
+    await request.get("/auth/logout", null).then(() => {
```

Comment:
Instead of giving `null` here and everywhere where a `GET` without parameter will be made, this parameter should be set to `null` by default in the wrapper I think.